### PR TITLE
adding support for installing the windows features required for iis

### PR DESCRIPTION
**Pull request by liamjbennett**

### CODE REVIEW

**Comment by ffrank** [comment]
Ping @liamjbennett should this be closed?

**Comment by juniorsysadmin** [comment]
@liamjbennett Close this PR?

**Comment by liamjbennett** [comment]
I'd actually like to keep get this in.

Do we think this requires a major version bump or not? Given that the init.pp at present is only there for example purposes we might be able to get away without it but I would like feedback first

File: manifests/features/application_development.pp [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      ensure_resource('windowsfeature', 'IIS-ISAPIFilter' )
+    }
+    '6.0','6.1': {
+      ensure_resource('windowsfeature', 'Web-Asp-Net' )
+      ensure_resource('windowsfeature', 'Web-Net-Ext' )
+      ensure_resource('windowsfeature', 'Web-ISAPI-Ext' )
+      ensure_resource('windowsfeature', 'Web-ISAPI-Filter' )
+    }
+    default: {
+      fail("Do not know how to install iis windows features for ${::kernalmajversion}")
```
Comment (juniorsysadmin) [inline manifests/features/application_development.pp]: typo

File: manifests/features/common_http.pp [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      ensure_resource('windowsfeature', 'IIS-HttpErrors' )
+      ensure_resource('windowsfeature', 'IIS-DefaultDocument' )
+    }
+    '6.0','6.1': {
+      ensure_resource('windowsfeature', 'Web-Static-Content' )
+      ensure_resource('windowsfeature', 'Web-Http-Errors' )
+      ensure_resource('windowsfeature', 'Web-Default-Doc' )
+    }
+    default: {
+      fail("Do not know how to install iis windows features for ${::kernalmajversion}")
```
Comment (juniorsysadmin) [inline manifests/features/common_http.pp]: typo

File: manifests/features/management_tools.pp [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    '6.2','6.3': {
+      ensure_resource('windowsfeature', 'IIS-WebServerManagementTools' )
+      ensure_resource('windowsfeature', 'IIS-ManagementConsole' )
+    }
+    '6.0','6.1': {
+      ensure_resource('windowsfeature', 'Web-Mgmt-Tools' )
+      ensure_resource('windowsfeature', 'Web-Mgmt-Console' )
+    }
+    default: {
+      fail("Do not know how to install iis windows features for ${::kernalmajversion}")
```
Comment (juniorsysadmin) [inline manifests/features/management_tools.pp]: typo

**Comment by juniorsysadmin** [comment]
> Do we think this requires a major version bump or not? Given that the init.pp at present is only there for example purposes we might be able to get away without it but I would like feedback first

Hmmm. If you're reasonably confident it doesn't change the behaviour with the same params, a major version bump isn't required. I'm a bit wary of the use of ```ensure_resource``` but that's me.

**Pull request merged**